### PR TITLE
[SOL-1946] New error endpoint

### PR DESCRIPTION
**Pull request by JoyceZhu**
FYI @mjfrey .

### CODE REVIEW

File: ecommerce/static/sass/partials/views/_cancel_error.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
+.receipt-cancel-error {
```
Comment (JoyceZhu) [inline ecommerce/static/sass/partials/views/_cancel_error.scss:1]: The cancel and error endpoints will use the same styling -- both pages only contain a short message about the order erroring out/being cancelled.

**Comment by JoyceZhu** [comment]
Screenshot of new error endpoint. Old LMS error endpoint can be viewed [here](https://coupons.sandbox.edx.org/commerce/checkout/error/)

screen shot 2016-07-11 at 11 15 05 am

**Comment by JoyceZhu** [comment]
@mjfrey @vkaracic @marjev could you please review this? Content is very similar to previous cancel checkout page PR.

File: ecommerce/templates/oscar/checkout/error.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    {% include 'edx/partials/_student_navbar.html' %}
+{% endblock navbar %}
+
+{% block content %}
+    <div class="container content-wrapper receipt-cancel-error">
+        <h1>{% trans "Checkout Error" %}</h1>
+        <p>{% blocktrans %} An error has occurred with your payment. <b>You have not been charged.</b>
+        {% endblocktrans %}</p>
+
+        <p>{% with "<a class='nav-link' href='mailto:"|add:payment_support_email|add:"'>"|safe as start_link %}
```
Comment (vkaracic) [inline ecommerce/templates/oscar/checkout/error.html:19]: Do you need the second `add` here, can you just leave the quote?

Comment (JoyceZhu) [inline ecommerce/templates/oscar/checkout/error.html:19]: I just tried out your suggestion @vkaracic -- unfortunately looks like I need the final `add`, as without the closing tag the email will be completely omitted from the generated HTML.

**Comment by vkaracic** [comment]
LGTM

File: ecommerce/extensions/checkout/views.py [hunk 3]
```diff
@@ -80,3 +80,30 @@ def get_context_data(self, **kwargs):
...
+
+class CheckoutErrorView(TemplateView):
+    """ Displays an error page when checkout does not complete successfully. """
+
+    template_name = 'checkout/error.html'
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, *args, **kwargs):
+        """
+        Request needs to be csrf_exempt to handle CyberSource's POST response.
```
Comment (mjfrey) [inline ecommerce/extensions/checkout/views.py]: Remove reference to Cybersource.  This is just generic support for a POST back from a payment processor.

File: ecommerce/extensions/checkout/views.py [hunk 4]
```diff
@@ -80,3 +80,30 @@ def get_context_data(self, **kwargs):
...
+        """
+        return super(CheckoutErrorView, self).dispatch(*args, **kwargs)
+
+    def post(self, request, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Allow CyberSource's post back.
```
Comment (mjfrey) [inline ecommerce/extensions/checkout/views.py]: Same here.

File: ecommerce/templates/oscar/checkout/error.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            {% blocktrans with end_link="</a>"|safe %}
+            Please try to submit your payment again. If this problem persists, contact {{ start_link }}
+            {{ payment_support_email }}{{ end_link }}.
+            {% endblocktrans %}
+        {% endwith %}</p>
+    
+    </div>
+{% endblock content %}
```
Comment (mjfrey) [inline ecommerce/templates/oscar/checkout/error.html]: Add a new line

**Comment by mjfrey** [comment]
👍 Looks good.
Just rebase on master.

**Pull request merged**